File: app/src/main/java/com/android/learn/mcontract/KnowledgeContract.java
package com.android.learn.mcontract;



import com.android.base.mmodel.TreeBean;

import java.util.List;


public class KnowledgeContract {

    public interface View  {
        void showKnowledge(List<TreeBean> data);

    }

    public interface Presenter {
        void getKnowledge();

    }

}

File: app/src/main/java/com/android/learn/mcontract/KnowledgeContract.kt
package com.android.learn.mcontract


import com.android.base.mmodel.TreeBean
import com.android.base.mview.BaseView


class KnowledgeContract {

    interface View : BaseView {
        fun showKnowledge(data: List<TreeBean>)

    }

    interface Presenter {
        fun getKnowledge()

    }

}
